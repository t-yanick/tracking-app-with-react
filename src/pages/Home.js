import React from 'react';
import { connect } from 'react-redux'
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import LogoutBtn from '../components/LogoutBtn';

const Home = ({ history, loginUser }) => (
  <div>
    <h1 className="heading">Home</h1>
    <div className="content">
      <h2 className="home__heading">
        Welcome to Track it App
      </h2>
      {loginUser ? <LogoutBtn history={history} /> : (
        <div>
          <Link to="/login" className="btn mb2 dark">Login</Link>
          <Link to="/signup" className="btn mb2 medium">Signup</Link>
        </div>
      )}
    </div>
  </div>
);

const mapStateToProps = (state) => ({
  loginUser: state.user.logIn,
});

Home.propTypes = {
  history: PropTypes.instanceOf(Object),
  loginUser: PropTypes.bool.isRequired,
};

Home.defaultProps = {
  history: null,
};

export default connect(mapStateToProps)(Home);