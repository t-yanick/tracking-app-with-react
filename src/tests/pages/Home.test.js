import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../../pages/Home';
import createTestStore from '../fixtures/createTestStore';

describe('Home', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Home />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
