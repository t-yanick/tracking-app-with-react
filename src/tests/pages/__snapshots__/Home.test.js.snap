// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match with snapshot 1`] = `
<div>
  <h1
    className="heading"
  >
    Home
  </h1>
  <div
    className="content"
  >
    <h2
      className="home__heading"
    >
      Welcome to Track it App
    </h2>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Logout
    </button>
  </div>
</div>
`;
